src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/9.diff.sql

```diff
@@ -0,0 +1,1875 @@
+/*************************************************************
+    This migration introduces table partitioning by ResourceTypeId on the Resource and all search parameter tables.
```

Comment:
I am not familiar with partitioned tables, but the [docs](https://docs.microsoft.com/en-us/sql/relational-databases/partitions/partitioned-tables-and-indexes?view=sql-server-ver15#benefits-of-partitioning) mention that setting `LOCK_ESCALATION = AUTO` can improve performance. Would it be helpful for us to do that?

Reply:
Great catch! Thanks.

Reply:
Added in https://github.com/microsoft/fhir-server/pull/1876/commits/c0d33abae9b6358959fb104c8e7108ce01a8b4b8